### PR TITLE
Implement local storage

### PR DESCRIPTION
**Pull request by rsasch**
see #458 for details

### CODE REVIEW

File: ui/src/app/job-list/job-list.component.ts [hunk 1]
```diff
@@ -75,6 +81,19 @@ export class JobListComponent implements OnInit {
...
     });
+
+    // set project ID (if any) and get display field info for list columns
+    const req = URLSearchParamsUtils.unpackURLSearchParams(this.route.snapshot.queryParams['q']);
+    this.projectId = req.extensions.projectId || '';
+    const savedColumnSettings = this.settingsService.getDisplayColumns(this.projectId);
+
+    // assign this.displayFields to a copy of this.capabilities.displayFields and then
+    // update with saved settings, if any
+    this.capabilities.displayFields.forEach((df) => this.displayFields.push(Object.assign({}, df)));
```
Comment (calbach) [inline ui/src/app/job-list/job-list.component.ts]: `map` would probably be more natural here.

```
this.displayFields = this.capabilities.displayFields.map((df) => Object.assign({}, df));
```

Comment (calbach) [inline ui/src/app/job-list/job-list.component.ts]: Nice find on `Object.assign()`

File: ui/src/app/job-list/job-list.component.ts [hunk 2]
```diff
@@ -151,6 +170,26 @@ export class JobListComponent implements OnInit {
...
+    this.displayFields.forEach((df) => {
+      if (df.field == field.field) {
+        df.primary = newValue;
+      }
+    })
+  }
+
+  saveSettings() {
+    let fields: string[] = [];
+    this.displayFields.forEach((field) => {
```
Comment (calbach) [inline ui/src/app/job-list/job-list.component.ts]:
```
fields = this.displayFields.filter(...)
```

Comment (rsasch) [reply ui/src/app/job-list/job-list.component.ts]: How would that work? `fields` is an array of string and `this.displayFields` is an array of DisplayField.

Comment (calbach) [reply ui/src/app/job-list/job-list.component.ts]: Good point, I missed that it was pulling out a subfield. filter + map would work, but I'm not sure I would recommend that over your current approach.

**Pull request merged**